src/libcore/option.rs

```diff
@@ -642,7 +643,7 @@ impl<T> Option<T> {
     }
 }
 
-impl<'a, T: Clone> Option<&'a T> {
+impl<'a, T: Clone, D: Deref<Target = T>> Option<D> {
     /// Maps an `Option<&T>` to an `Option<T>` by cloning the contents of the
     /// option.
     #[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
I'm agree, but for doc `Option<&T>` looks simple to understand method's purpose. Do you have any idea how to describe it? `Option<Deref>` is little unclear.